### PR TITLE
CryptoPkg/BaseCryptLib: add additional RSAES-OAEP crypto functions

### PR DESCRIPTION
**Pull request by cmruffin**
## Description
Expand the availability of the RSAES-OAEP crypto capability in
BaseCryptLib.  Applications using RSA crypto functions directly from
OpensslLib can transition to BaseCryptLib to take advantage of the
shared crypto feature in CryptoDxe.

- [ ] Impacts functionality?
- [x] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
- [ ] Includes documentation?

## How This Was Tested
Host-based unit tests, end-to-end testing with shared crypto binary.

## Integration Instructions
Depends on implementation in mu_crypt_release/CryptoBinPkg/OpensslLib.

### CODE REVIEW

**Comment by cmruffin** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Review by makubacki — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

**Comment by cmruffin** [comment]
typo: rsaes-oaep

**Review by kenlautner — APPROVED** [review]

File: CryptoPkg/CryptoPkg.dec [hunk 1]
```diff
@@ -81,212 +81,215 @@
...
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs7GetSigners|FALSE|BOOLEAN|0x13
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs7FreeSigners|FALSE|BOOLEAN|0x14
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs7GetCertificatesList|FALSE|BOOLEAN|0x15
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs7Sign|FALSE|BOOLEAN|0x16
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs7Verify|FALSE|BOOLEAN|0x17
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServiceVerifyEKUsInPkcs7Signature|FALSE|BOOLEAN|0x18
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs7GetAttachedContent|FALSE|BOOLEAN|0x19
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServiceAuthenticodeVerify|FALSE|BOOLEAN|0x1a
-  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServiceImageTimestampVerify|FALSE|BOOLEAN|0x1b
+  gEfiCryptoPkgTokenSpaceGuid.PcdCryptoServicePkcs1v2Decrypt|FALSE|BOOLEAN|0x13
```
Comment (kuqin12) [inline CryptoPkg/CryptoPkg.dec:84]: I might have missed something, but does this PCD really have to be inserted? It seems that some of the changes called are not that as necessary?

Comment (cmruffin) [reply CryptoPkg/CryptoPkg.dec:84]: Thanks for your comment.  It seems to be an artifact of the generate_cryptodriver.py design - it seems to assign PCD integers to PCDs in the same sequence it discovers them in the BaseCryptoLib.h.

**Comment by makubacki** [comment]
@cmruffin, please confirm you are ready for this PR to be merged and I'll do it.

**Comment by cmruffin** [comment]
> @cmruffin, please confirm you are ready for this PR to be merged and I'll do it.

Yes sir I’m all set.  Thank you!

**Comment by cmruffin** [comment]
Have to split this PR into 2 for integration.
1) first PR adds function prototypes and will allow PR in mu_crypto_release to go
2) second PR will add support to CryptoOnPpiProtocol Library and update Crypto binary extdep

**Comment by makubacki** [comment]
> Have to split this PR into 2 for integration.

@cmruffin and I spoke offline about ordering between this PR repo and the one in mu_crypto_release. The plan is that he will make some updates in this PR that will ultimately depend on a binary being generated in over there first.

**Comment by cmruffin** [comment]
@makubacki this PR has been updated with the -rc version binary

**Pull request merged**